transport/http/handler.go

```diff
@@ -93,8 +96,20 @@ func (h handler) callHandler(w http.ResponseWriter, req *http.Request) error {
 		ctx = baggage.NewContextWithHeaders(ctx, headers.Items())
 	}
 
+	start := time.Now()
+
 	// TODO capture and handle panic
-	return h.Handler.Handle(ctx, httpOptions, treq, newResponseWriter(w))
+	err = h.Handler.Handle(ctx, httpOptions, treq, newResponseWriter(w))
+
+	// The handler is well behaved and stopped work on context deadline. We
+	// forward this information to the client diligently.
+	if err != nil && err == context.DeadlineExceeded && err == ctx.Err() {
```

Comment:
Nit: Don't need to check `!= nil` if you're already checking `== context.DeadlineExceeded`